batch_writer.go

```diff
@@ -123,7 +125,12 @@ func (w *BatchWriter) WriteRowBatch(batch *RowBatch) error {
 		// Note that the state tracker expects us the track based on the original
 		// database and table names as opposed to the target ones.
 		if w.StateTracker != nil {
-			w.StateTracker.UpdateLastSuccessfulPaginationKey(batch.TableSchema().String(), endPaginationKeypos, uint64(batch.Size()))
+			bytesWrittenForThisBatch := uint64(0)
+			if w.enableRowBatchSize {
+				bytesWrittenForThisBatch = batch.EstimateByteSize()
+			}
+			w.StateTracker.UpdateLastSuccessfulPaginationKey(batch.TableSchema().String(), endPaginationKeypos,
+				RowStats{NumBytes: bytesWrittenForThisBatch, NumRows: uint64(batch.Size())})
```

Comment:
This behaviour seems a bit sketchy. If someone has turned off the `EnableRowBatchSize` to off they'll see the `NumBytes` to 0 which would create confusion. I don't think there's much we can do in here other then documenting this behaviour.